org.adempiere.base/src/org/compiere/model/MUser.java

```diff
@@ -1092,12 +1092,7 @@ protected boolean afterSave(boolean newRecord, boolean success) {
 				MPasswordHistory passwordHistory = new MPasswordHistory(this.getCtx(), 0, this.get_TrxName());
 				passwordHistory.setSalt(this.getSalt());
 				passwordHistory.setPassword(this.getPassword());
-				// http://wiki.idempiere.org/en/System_user
-				if (!this.is_new() && this.getAD_User_ID() == 0){
-					passwordHistory.set_Value(MPasswordHistory.COLUMNNAME_AD_User_ID, 0);
-				}else{
-					passwordHistory.setAD_User_ID(this.getAD_User_ID());
-				}
+				passwordHistory.setAD_User_ID(this.getAD_User_ID());
```

Comment:
Here is the line that exemplifies what I am not happy with. setAD_User_ID() now accepts zero as a valid value. Still getAD_User_ID() does not distinguish between a null and a zero in the column. It will in both cases return zero. A null that is read from one table (or that could be in place because of an initialization error)  gets written as a zero in the other table. This mechanism could potentially introduce or propagate an error that could not be spotted and might lead to  corrupt database entries. I'd suggest to also override the getAD_User_ID() method to remove the possible ambivalent state.